### PR TITLE
Ovl_En_Pametfrog OK with partial documentation

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Most of the functions have been named with 2 main exceptions:
- func_8086A2CC and func_8086A238 deal with CollisionPoly/Matrices I didn't quite understand
- Most functions related to damage effects. I couldn't quite discern which damage effect was what

Below is the flow of Pametfrogs. Any feedback for code or documentation is greatly appreciated!

En_Pametfrog gv

### CODE REVIEW

File: src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c [hunk 1]
```diff
@@ -21,144 +97,1295 @@ const ActorInit En_Pametfrog_InitVars = {
...
+    f32 sin;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->actor.shape.rot.x += 0x800;
+    this->actor.shape.rot.z += 0x1000;
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    sin = sin_rad(this->timer * 1.0471976f) * ((0.02f * (this->timer * 0.16666667f)) + 0.005f) + 1.0f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c:509]: Would this match?
```suggestion
    sin = sin_rad(this->timer * (M_PI/3.0f)) * ((0.02f * (this->timer * (1.0f/6.0f))) + 0.005f) + 1.0f;
```

File: src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c [hunk 2]
```diff
@@ -21,144 +97,1295 @@ const ActorInit En_Pametfrog_InitVars = {
...
+    Vec3f worldPos2;
+    f32 doubleSpeedXZ;
+    u32 bgId1;
+    u32 bgId2;
+    s32 isSuccess;
+
+    poly1 = NULL;
+    poly2 = NULL;
+    isSuccess = false;
+
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c:586]:
```suggestion
    CollisionPoly* poly1 = NULL;
    CollisionPoly* poly2 = NULL;
    Vec3f vec1;
    Vec3f vec2;
    Vec3f worldPos1;
    Vec3f worldPos2;
    f32 doubleSpeedXZ;
    u32 bgId1;
    u32 bgId2;
    s32 isSuccess = false;

```

File: src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c [hunk 3]
```diff
@@ -21,144 +97,1295 @@ const ActorInit En_Pametfrog_InitVars = {
...
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnPametfrog_LookAround;
+}
+
+void EnPametfrog_LookAround(EnPametfrog* this, GlobalContext* globalCtx) {
+    EnPametfrog_IdleWaterEffects(this, globalCtx);
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 5, 0x400, 0x80);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) && (func_801690CC(globalCtx) == 0)) {
+        if (this->unk_2AE == false) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c:1020]:
```suggestion
        if (!this->unk_2AE) {
```

**Review by AngheloAlf — COMMENTED** [review]
Good work!
I leaved a few comments.
Also, could you try to look at those weird floats and try to convert them to something more readable? I left some examples that may or may not match.

File: src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c [hunk 4]
```diff
@@ -21,144 +97,1295 @@ const ActorInit En_Pametfrog_InitVars = {
...
+
+void EnPametfrog_RearOnSnapperRise(EnPametfrog* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->timer--;
+    if (this->timer == 0) {
+        EnPametfrog_SetupRearOnSnapper(this);
+    } else {
+        this->actor.world.pos.y =
+            Math_SinS(this->timer * 0xCCC) * 100.0f + (((EnBigpamet*)this->actor.child)->unk_2AC + 46.0f);
+    }
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c:474]: Doesn't seem like it, no :(

File: src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c (hunk 1 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Pametfrog/z_en_pametfrog.c]: Sure does!

**Review by jpburnett — APPROVED** [review]
Really good work with this, looking over it I didn't see anything that stood out. Once more progress is made I think documenting this could be more feasible.

**Pull request merged**